src/cider/nrepl/middleware/stacktrace.clj

```diff
@@ -128,14 +128,17 @@
   "In order to match more namespaces, we look for a common namespace
   prefix across the ones we have identified."
   (let [common
-        (try (reduce
-              (fn [common ns]
-                (let [ns (str/lower-case ns)
-                      matched (map vector common ns)
-                      coincident (take-while (fn [[a b]] (= a b)) matched)]
-                  (apply str (map first coincident))))
-              (str/lower-case (first directory-namespaces))
-              directory-namespaces)
+        (try (->> directory-namespaces
+                  ;; remove usually present namespaces
+                  ;; that would break the commonality:
+                  (remove #{'dev 'user})
```

Comment:
Maybe this can be generalized to "single-segment namespace", for covering unforeseen patterns as well.

I think I see other single-segment namespaces from time to time e.g. `test-runner` or `test-setup`

Reply:
I agree. It's better to make this more robust.